lib/beacon/css_compiler.ex

```diff
@@ -0,0 +1,65 @@
+defmodule Beacon.CSSCompiler do
+  @moduledoc """
+  Default CSS compiler for the runtime CSS compilation.
+  """
+
+  alias Beacon.Components
+  alias Beacon.Layouts.Layout
+  alias Beacon.Pages
+
+  @behaviour Beacon.RuntimeCSS
+
+  @impl Beacon.RuntimeCSS
+  def compile!(%Layout{} = layout, opts \\ []) do
+    tmp_dir = System.tmp_dir!()
+    config_file = Path.join(tmp_dir, "tailwind.config.js")
+    output_file = Path.join(tmp_dir, "runtime.css")
+    input_file = Path.join([:code.priv_dir(:beacon), "assets/beacon.css"])
+
+    raw_content = [layout.body, page_templates(layout.id), component_bodies()]
+
+    config = build_config(opts[:config_template], raw_content) |> IO.inspect()
+
+    File.write!(config_file, config)
+
+    0 = Tailwind.run(:runtime, ~w(
+      --config=#{config_file}
+      --input=#{input_file}
+      --output=#{output_file}
+      --minify
+    ))
+
+    output = File.read!(output_file)
+
+    cleanup!(config_file, output_file)
```

Comment:
Just food for thought but we could leverage custom exceptions or raise with a custom message to expose better feedback, instead of returning something like MatchError or File.Error. Wdyt?

Reply:
Just to be clear, the current code looks good and there's no need to change it now :)

But eventually we'll need to provide feedback to users depending on the situation, similar to https://github.com/BeaconCMS/beacon/pull/35#issuecomment-1264392147

So we can move forward to get this PR merged and later discuss approaches to handle compilation errors. Makes sense?

Reply:
Generic runtime error with a bit of a better message for now: https://github.com/BeaconCMS/beacon/pull/30/files#diff-d12b7007afa8a07707b6bf598cbb07d910acdbe84669cf03e27fdf5b75e1413fR20-R36

Reply:
Thanks! That's gonna help with debugging and identifying where we have to deal with compilation errors.